raiden/tests/integration/rpc/assumptions/test_geth_rpc_assumptions.py

```diff
@@ -26,7 +25,14 @@ def send_transaction():
     # geth keeps the latest 128 blocks before pruning. Unfortunately, this can
     # not be configured to speed this test up.
     non_pruned_blocks = 128
-    while web3.eth.blockNumber < pruned_block_number + non_pruned_blocks + 1:
+    # According to the geth devs if we run a PoA chain (clique, dev mode) and
+    # HEAD-127 doesn't contain any transactions, then the state of HEAD-127 will
+    # be the same as HEAD-128, so it will still be referenced and not deleted.
+    # To avoid this and make sure the pruned_block_number is pruned let's mine
+    # buffer period more blocks while at the same time sending transactions
+    buffer_period_blocks = 5
+    target_block = pruned_block_number + non_pruned_blocks + buffer_period_blocks + 1
```

Comment:
If the above is correct, then this whole test is overly complicated. We can just do

Mine a transaction, get the block number in which it was mined from the receipt, wait for this transaction to be in the 127th block, and the query for the *previous block*. Because of the first statement on the other comment, we know the 128th will be pruned, since the 127th has a transaction.

Reply:
Yes this makes sense ... assuming what they say is correct.